engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -0,0 +1,321 @@
+        let block_hash = get_hash(&row, "block_hash");
+        let block_timestamp = get_timestamp(&row, "timestamp");
+        let index: i32 = row.get("index");
+        let id: i64 = row.get("id");
+        let hash = get_hash(&row, "hash");
+        let near_hash = get_hash(&row, "near_hash");
+        let near_receipt_hash = get_hash(&row, "near_receipt_hash");
+        let from = get_address(&row, "from");
+        let to: Option<&[u8]> = row.get("to");
+        let nonce = get_numeric(&row, "nonce");
+        let gas_price = get_numeric(&row, "gas_price");
+        let gas_limit = get_numeric(&row, "gas_limit");
+        let gas_used = get_numeric(&row, "gas_used");
+        let value = get_numeric(&row, "value");
+        let input: Option<Vec<u8>> = row.get("input");
+        let v = get_numeric(&row, "v");
+        let r = get_numeric(&row, "r");
+        let s = get_numeric(&row, "s");
+        let status: bool = row.get("status");
+        let output: Option<Vec<u8>> = row.get("output");
+
+        Self {
+            block: block as u64,
+            block_hash,
+            block_timestamp: block_timestamp.unwrap(),
+            index: index as u16,
+            id: id as u64,
+            hash,
+            near_hash,
+            near_receipt_hash,
+            from,
+            to: to.map(Address::from_slice),
+            nonce,
+            gas_price,
+            gas_limit,
+            gas_used: gas_used.low_u64(),
+            value: Wei::new(value),
+            input: input.unwrap_or_else(Vec::new),
+            v: v.low_u64(),
+            r,
+            s,
+            status,
+            output: output.unwrap_or_else(Vec::new),
+        }
+    }
+}
+
+impl From<TransactionRow> for EthTransactionKind {
+    fn from(row: TransactionRow) -> Self {
+        let legacy = LegacyEthSignedTransaction {
+            transaction: TransactionLegacy {
+                nonce: row.nonce,
+                gas_price: row.gas_price,
+                gas_limit: row.gas_limit,
+                to: row.to,
+                value: row.value,
+                data: row.input,
+            },
+            v: row.v,
+            r: row.r,
+            s: row.s,
+        };
+
+        Self::Legacy(legacy)
+    }
+}
+
+fn get_numeric(row: &postgres::Row, field: &str) -> U256 {
+    let value: PostgresNumeric = row.get(field);
+    U256::try_from(value).unwrap()
+}
+
+fn get_hash(row: &postgres::Row, field: &str) -> H256 {
+    let value: &[u8] = row.get(field);
+    H256::from_slice(value)
+}
+
+fn get_address(row: &postgres::Row, field: &str) -> Address {
+    let value: &[u8] = row.get(field);
+    Address::from_slice(value)
+}
+
+fn get_timestamp(row: &postgres::Row, field: &str) -> Option<u64> {
+    let timestamp: Option<SystemTime> = row.get(field);
+    timestamp
+        .and_then(|t| t.duration_since(SystemTime::UNIX_EPOCH).ok())
+        .map(|d| d.as_nanos() as u64)
+}
+
+struct PostgresNumeric {
```

Comment:
The postgres library for rust didn't include a way to parse the `Numeric` type, so I had to write my own.